stac_fastapi/elasticsearch/core.py

```diff
@@ -154,7 +155,15 @@ def get_queryset(self, **kwargs) -> Search:
 
                 qs = qs.query(filter)
 
-        return qs
+        matched = qs.count()
+        if self.extension_is_enabled('ContextExtension'):
+            context = {
+                'returned':limit if page*limit <= matched-1 else (matched-1)-(page-1)*limit,
+                'limit':limit,
+                'matched':matched,
+            }
```

Comment:
Count returns the same value before and after pagination filter. So returned has to be calculated.